<launch>

  <!-- Parameters possible to change-->
  <arg name="num_sensors"           default="8" />
  <arg name="radius"                default="4.0" />
  <arg name="num_samples"           default="80" />
  
  <arg name="rows"                  default="6" />
  <arg name="cols"                  default="5" />
  <arg name="cell_width"            default="0.12" />
  <arg name="cell_height"           default="0.12" />
  
  <arg name="camera_name"           default="sim_camera" />
  <arg name="camera_calib_url"      default="file://$(find rgbd_calibration)/conf/$(arg camera_name).yaml" />
  
  <arg name="image_error"           default="0.5" />
  
  <arg name="height"                default="2.0" />
  <arg name="delta_radius"          default="0.5" />
  
  <arg name="results_file"          default="/tmp/results_circle.txt" />

<!-- Openning Rviz for visualization-->
	<!--node name="rviz" pkg="rviz" type="rviz" args="-d $(env HOME)/.rviz/multikinect.rviz"/-->

<!-- Launching calibration -->
	<node pkg="multicamera_calibration" type="multicamera_simulation" name="multicamera_simulation" output="screen">
	
	  <param name="type"            value="circle" />
	
	  <param name="num_sensors"      value="$(arg num_sensors)" />
	  <param name="radius"           value="$(arg radius)" />
	  <param name="num_samples"      value="$(arg num_samples)" />
	  
	  <param name="rows"             value="$(arg rows)" />
      <param name="cols"             value="$(arg cols)" />
      <param name="cell_width"       value="$(arg cell_width)" />
      <param name="cell_height"      value="$(arg cell_height)" />
      
      <param name="camera_name"      value="$(arg camera_name)" />
      <param name="camera_calib_url" value="$(arg camera_calib_url)" />
      
      <param name="image_error"      value="$(arg image_error)" />
      
      <param name="height"           value="$(arg height)" />
      <param name="delta_radius"     value="$(arg delta_radius)" />
	  
	  <param name="results_file"     value="$(arg results_file)" />
	  
	</node>

</launch>
